libs/constrained-generators/testlib/Test/Min/Model.hs

```diff
@@ -0,0 +1,1306 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE FunctionalDependencies #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE PatternSynonyms #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE ViewPatterns #-}
+-- HasSpec instances for known types Integer, Bool, Set , (,)
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Test.Min.Model where
+
+import Constrained.Core (
+  Evidence (..),
+  Var (..),
+  eqVar,
+  freshen,
+  unionWithMaybe,
+ )
+import Constrained.Env
+import Constrained.GenT
+import qualified Constrained.Graph as Graph
+import Constrained.List hiding (ListCtx)
+import Control.Monad (guard)
+import Control.Monad.Writer (Writer, runWriter, tell)
+import Data.Foldable (fold)
+import qualified Data.Foldable as Foldable (fold)
+import Data.Kind
+import Data.List (nub, partition, (\\))
+import Data.List.NonEmpty (NonEmpty (..))
+import qualified Data.List.NonEmpty as NE
+import Data.Maybe (isNothing, listToMaybe, maybeToList)
+import Data.Semigroup (Any (..))
+import Data.Set (Set)
+import qualified Data.Set as Set
+import Data.Typeable
+import GHC.Stack
+import Prettyprinter
+import Test.Min.Base
+import Test.Min.Syntax
+import Test.QuickCheck hiding (forAll)
+
+-- ====================================================
+-- Now some concrete examples
+-- 1) Introduce the function symbols
+-- 2) Give the Syntax, Semantics, and Logic instances
+-- 3) Give the HasSpec instance
+-- ====================================================
+
+-- ======== Integer example ==============
+
+data IntegerSym (dom :: [Type]) rng where
+  PlusW :: IntegerSym '[Integer, Integer] Integer
+  MinusW :: IntegerSym '[Integer, Integer] Integer
```

Comment:
What about `Negate` instead?

Reply:
That would give us more examples of different arities

Reply:
Didn't think about that. But we do use both (+.) and (-.) in the some examples.

Reply:
go back to negate

Reply:
So I added negate